tests/activity_sticky_queues/activity_sticky_worker_workflow_test.py

```diff
@@ -0,0 +1,167 @@
+                mock_download,
+                mock_work,
+                mock_cleanup,
+                tasks.get_available_task_queue,
+            ],
+        ):
+            await client.execute_workflow(  # noqa
+                tasks.FileProcessing.run,
+                id=str(uuid.uuid4()),
+                task_queue=task_queue_name,
+                retry_policy=retry_policy,
+                run_timeout=run_timeout,
+            )
+
+
+async def test_processing_fails_gracefully(client: Client):
+    task_queue_name = str(uuid.uuid4())
+
+    @activity.defn(name="get_available_task_queue")
+    async def mock_task_queue():
+        return task_queue_name
+
+    async with Worker(
+        client,
+        task_queue=task_queue_name,
+        workflows=[tasks.FileProcessing],
+        activities=[
+            mock_download,
+            mock_work_fail,
+            tasks.clean_up_file_from_worker_filesystem,
+            mock_task_queue,
+        ],
+    ):
+        exception_catch = pytest.raises(WorkflowFailureError)
+        with exception_catch, mock.patch.object(tasks, "delete_file") as mock_del:
+            await client.execute_workflow(
+                tasks.FileProcessing.run,
+                id=str(uuid.uuid4()),
+                task_queue=task_queue_name,
+                retry_policy=RetryPolicy(
+                    maximum_attempts=1,
+                    maximum_interval=timedelta(milliseconds=50),
+                ),
+                run_timeout=timedelta(milliseconds=50),
+            )
+        mock_del.assert_called_once()
+        # assert output == sentinel_value
+
+
+async def test_worker_runs(client: Client):
+    task_queue_name_distribution = str(uuid.uuid4())
+    task_queue_workers = [str(uuid.uuid4()) for _ in range(3)]
+
+    @activity.defn(name="get_available_task_queue")
+    async def mock_task_queue():
+        return random.choice(task_queue_workers)
+
+    worker_distribution = Worker(
+        client,
+        task_queue=task_queue_name_distribution,
+        workflows=[tasks.FileProcessing],
+        activities=[
+            mock_task_queue,
+        ],
+    )
+    worker_1, worker_2, worker_3 = [
+        Worker(
+            client,
+            task_queue=queue,
+            activities=[
+                mock_download,
+                mock_work,
+                mock_cleanup,
+            ],
+        )
+        for queue in task_queue_workers
+    ]
+
+    # Should execute all fine
+    # p(pass| should fail) = 0.333 ^ (n_workers * n_iters)~ 6e-8
+    workflow_ids = [str(uuid.uuid4()) for _ in range(5)]
+    async with worker_distribution, worker_1, worker_2, worker_3:
+        for workflow_id in workflow_ids:
+            result = await client.execute_workflow(
+                tasks.FileProcessing.run,
+                id=workflow_id,
+                task_queue=task_queue_name_distribution,
+                retry_policy=retry_policy,
+                run_timeout=run_timeout,
```

Comment:
Should not set these values unless you really need them

Reply:
These were included due to `should fail/does fail` leaving the test server hanging - recommended implementation to keep it moving?

Reply:
These were included due to `should fail/does fail` leaving `poe test` hanging. The test cases pass (with warnings) but the test server never cleans up. What's the recommended setup to prevent the test server from choking on deliberately failed test cases? Pushing up removal now in case you need a local replication

Reply:
:+1: Will investigate

Reply:
Just ran the tests, and it seems it's because the `get_available_task_queue` activity retries forever with a `NotImplementedError`. By default activities will retry forever, and that can halt a worker shutdown. At a quick glance it seems `test_workflow_fails_without_task_queue_fn` uses the not-implemented `get_available_task_queue` which retries forever on its error. I can investigate a bit deeper later if that's not clear.